### ISSUE TITLE
ListPodSandbox: bad wiretype for field runtime.PodSandboxFilter.State

### ISSUE THREAD

**Issue by jjlakis** [kubernetes-incubator/cri-o#339]
Problem appeared when running local-up-cluster from current kubernetes master branch (`28c439dcfb03832044469f035319daf53d60267e`) with current cri-o master. 
Hope it's OK to describe it using my standard debug print that prints which GRPC request is passed to `ocid` and what is the result of this request: https://github.com/intelsdi-x/cri-o/commit/ab854dde116c3a00b1ea6c7cd0f97abcfc275d19

Steps to reproduce:
1. Start ocid and local cluster.
```
$ sudo /home/go/bin/ocid --runtime $(which runc) --log /home/jlakis/ocid.log --debug
# EXPERIMENTAL_CRI=true LOG_LEVEL=6 CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT=/var/run/ocid.sock ./hack/local-up-cluster.sh -O
```

During first 1-2 minutes of running cluster, `ListPodSandbox` is failing sometimes: 
```
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  OK
GRPC query -->  ListContainers
GRPC query -->  ListContainers  OK
GRPC query -->  Status
GRPC query -->  Status  OK
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  OK
GRPC query -->  ListContainers
GRPC query -->  ListContainers  OK
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  OK
GRPC query -->  ListContainers
GRPC query -->  ListContainers  OK
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
```

After that, ListPodSandbox is failing more often (actually it is passing sometimes):

```
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox  OK
GRPC query -->  ListContainers
GRPC query -->  ListContainers  OK
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0
GRPC query -->  ListPodSandbox
GRPC query -->  ListPodSandbox  ERROR:  proto: bad wiretype for field runtime.PodSandboxFilter.State: got wiretype 2, want 0

```

What is expected:
Passing `ListPodSandbox` request.

cc @mzylowski @runcom @sameo

**Comment by runcom** [comment]
weird, @nalind could you also take a look? seems like, at some point, we start returning some bad state from sandboxes (which might be related to storage? just wondering)

**Comment by mzylowski** [comment]
After my short check: error requests even do not enter to `ListPodSandbox` method in sandbox_list.go, the problem is somewhere else.

**Comment by mrunalp** [comment]
Is there a protobuf diff between what we have vendored and upstream? If so, we can try to bump up our version.

**Comment by mzylowski** [comment]
@mrunalp I came yesterday to do the same conclusions.
8 days ago this was merged into k8s master: https://github.com/kubernetes/kubernetes/pull/39158
So k8s try to talk with cri-o server via new API.
I tried to bump k8s version via running hack/vendor.sh
After that is problem with building vendor package: github.com/containers/image because some files are now in k8s.io/apimachinery. To workaround this I just copy missing files (pkg/utils/net and pkg/utils/errors) . This commit is related: https://github.com/kubernetes/kubernetes/commit/67fac70ea307c04487027062c0f669883a6ecee2

After that I realized there will be a lot of changes in server side to work with new kubelet api. I started some changes and published it in https://github.com/kubernetes-incubator/cri-o/pull/342

@runcom @mrunalp Should I continue my work with updating server side to new api version?

**Comment by runcom** [comment]
That would be great, we're trying to solve the issues in containers/image as we speak though

**Comment by mzylowski** [comment]
Hmmm I'm not sure why I came across vendor problem in containers/image. This commit should fix it: https://github.com/containers/image/commit/6cb9b80664939f7235eca93748781ffa75df7853

**Comment by runcom** [comment]
@mzylowski this should also fix this issue with c/image https://github.com/containers/image/pull/223 but we're actually stuck with some issues with vendoring OCI/image-spec

**Comment by mzylowski** [comment]
I described my problem in more details here: https://github.com/kubernetes-incubator/cri-o/issues/343
I created new issue because with @jjlakis we think there is additional problem with vendoring script.
For now we will continue solving #342 with manually copied missing packages.

**Comment by mzylowski** [comment]
After #342 this can be closed.
